src/utils/utils.cpp

```diff
@@ -668,42 +668,48 @@ QueryData ExecuteQuery(mg_session *session, const std::string &query) {
     }
   }
 
-  {
-    const mg_map *summary = mg_result_summary(result);
-    for (size_t i = 0; i < mg_map_size(summary); i++) {
-      const mg_string *key = mg_map_key_at(summary, i);
-
-      // Parse only stats and notifications from summary
-      if (strcmp(mg_string_data(key), "stats") == 0) {
-        ret.stats = std::map<std::string, int64_t>{};
-        const mg_map *stats_map = mg_value_map(mg_map_value_at(summary, i));
-
-        for (size_t j = 0; j < mg_map_size(stats_map); j++) {
-          const auto *stat_key = mg_string_data(mg_map_key_at(stats_map, j));
+  const mg_map *summary = mg_result_summary(result);
+  if (summary && mg_map_size(summary) > 0) {
+    {
+      // parse stats from the summary
+      const mg_value *mg_stats = mg_map_at(summary, "stats");
+      if (mg_stats) {
+        const mg_map *stats = mg_value_map(mg_stats);
+        ret.stats = std::map<std::string, std::int64_t>{};
+        for (size_t j = 0; j < mg_map_size(stats); j++) {
+          const mg_string *mg_stat_key = mg_map_key_at(stats, j);
+          const auto stat_key = std::string(mg_string_data(mg_stat_key),
+                                            mg_string_size(mg_stat_key));
 
           const int64_t stat_value =
-              mg_value_integer(mg_map_value_at(stats_map, j));
+              mg_value_integer(mg_map_value_at(stats, j));
           ret.stats->insert({stat_key, stat_value});
         }
-      } else if (strcmp(mg_string_data(key), "notifications") == 0) {
+      }
+    }
+    {
+      // parse notifications from the summary
```

Comment:
I would split this function into two functions  `ParseNotificationsFromSummary` and  `ParseStatsFromSummary`.